seed/analysis_pipelines/pipeline.py

```diff
@@ -31,49 +31,166 @@ def task_create_analysis_property_views(analysis_id, property_view_ids, progress
         progress_data.step('Copying property data')
     analysis_view_ids, failures = AnalysisPropertyView.batch_create(analysis_id, property_view_ids)
     for failure in failures:
+        truncated_user_message = f'Failed to copy property data for PropertyView ID {failure.property_view_id}: {failure.message}'
+        if len(truncated_user_message) > 255:
+            truncated_user_message = truncated_user_message[:252] + '...'
         AnalysisMessage.objects.create(
             analysis_id=analysis_id,
             type=AnalysisMessage.DEFAULT,
-            user_message=f'Failed to copy property data for PropertyView ID {failure.property_view_id}: {failure.message}',
+            user_message=truncated_user_message,
         )
     return analysis_view_ids
 
 
-def check_analysis_status(status):
-    """Decorator factory for checking the status of the analysis, and raise an
-    exception if the status isn't what was expected
-
-    This can be used as a guard to avoid running a task when an analysis has been
-    stopped or failed elsewhere.
-
-    :param status: int, one of Analysis.STATUS_TYPES
+def analysis_pipeline_task(expected_status):
+    """Decorator factory for analysis pipeline celery tasks. In other words, this
+    function _returns_ a decorator for wrapping tasks used in analysis pipelines.
+    This wrapper **MUST** be added to all pipeline tasks to avoid unexpected errors.
+    The function and its decorators must look something like this:
+    ```
+    @shared_task(bind=True)
+    @analysis_pipeline_task(Analysis.SOME_STATUS)
+    def my_task(self, analysis_id, ...):
+        ...
+    ```
+    Note:
+    - `bind=True` is provided for the shared_task decorator (necessary for stopping the celery chain, ie when we don't want to run the remaining tasks)
+    - `self` is the first argument of the function (the celery task instance is passed as first argument when `bind=True`)
+    - `analysis_id` is some argument (it can be in any position)
```

Comment:
I thought `analysis_id` would be required to verify existence as well as status.

Reply:
That's correct. Did you see something that contradicts that? Or should I add that note?